src/python/pants/engine/legacy/graph.py

```diff
@@ -461,21 +481,6 @@ class SourcesSnapshots(Collection[SourcesSnapshot]):
     """
 
 
-@dataclass(frozen=True)
-class TopologicallyOrderedTargets:
-    """A set of HydratedTargets, ordered topologically from least to most dependent.
-
-    That is if B depends on A then B follows A in the order.
-
-    Note that most rules won't need to consider target dependency order, as those dependencies
-    usually implicitly create corresponding rule graph dependencies and per-target rules will then
-    execute in the right order automatically. However there can be cases where it's still useful for
-    a single rule invocation to know about the order of multiple targets under its consideration.
-    """
-
-    hydrated_targets: HydratedTargets
-
-
```

Comment:
Benjy gave permission to remove this all. The implementation was slightly incorrect due to treating `HydratedTarget.dependencies` having type `Tuple[HydratedTarget, ...]`, when it really was `Tuple[Address, ...]`. We don't use this rule anywhere, so it was much simpler to remove than to fix this and the tests.

(If we need this in the future, we can always retrace our steps thanks to Git.)